bin/rmt-data-import.rb

```diff
@@ -0,0 +1,202 @@
+        end
+      end
+    end
+  end
+
+  def import_systems
+    ActiveRecord::Base.transaction do
+      read_csv('systems').each do |row|
+        login, password, hostname = row
+
+        # rubocop:disable Rails/TimeZone
+        System.create!(
+          login: login,
+          password: password,
+          hostname: hostname,
+          registered_at: Time.now # FIXME: preserve registration date?
+        )
+        # rubocop:enable Rails/TimeZone
+        puts "Imported system #{login}"
+      end
+    end
+  end
+
+  def import_activations
+    ActiveRecord::Base.transaction do
+      read_csv('activations').each do |row|
+        login, product_id = row
+
+        product = Product.find_by(id: product_id)
+        system = System.find_by(login: login)
+
+        if !system
+          warn "System #{login} not found"
+          next
+        elsif !product
+          warn "Product #{product_id} not found"
+          next
+        else
+          activation = Activation.find_by(system: system, service: product.service)
+          unless activation
+            Activation.create!(system: system, service: product.service)
+            puts "Imported activation of #{product_id} for #{login}"
+          end
+        end
+      end
+    end
+  end
+
+  def import_hardware_info
+    systems = {}
+
+    # create hardware infos
+    read_csv('hw_info').each do |row|
+      login, key, value = row
+      next unless key == 'machinedata'
+
+      systems[login] = JSON.parse(value)['hwinfo']
+    end
+
+    ActiveRecord::Base.transaction do
+      systems.each do |login, info|
+        system = System.find_by(login: login)
+
+        unless system
+          warn "System #{login} not found"
+          next
+        end
+        info.delete('hostname')
+
+        HwInfo.find_or_initialize_by(system: system).update!(info)
+        puts "Hardware information stored for system #{login}"
+      end
+    end
+  end
+
+  def run(argv)
+    parse_cli_arguments argv
+    check_products_exist
+
+    import_repositories
+    import_custom_repositories
+
+    exit 0 if config.no_systems
+
+    import_systems
+    import_activations
+    import_hardware_info
+  end
+
+  def parse_cli_arguments(argv)
+    parser = OptionParser.new do |parser|
+      parser.on('-d', '--data PATH', 'Path to unpacked SMT data tarball') { |path| config.data_dir = path }
+      parser.on('--no-systems', 'Import no systems to rmt')               { config.no_systems = true }
+    end
+    parser.parse!(argv)
+    raise OptionParser::MissingArgument if config.data_dir.nil?
+  rescue OptionParser::InvalidOption, OptionParser::MissingArgument
+    puts parser
+    exit
+  end
+
+  def check_products_exist
+    return if Product.count > 0
+
+    warn 'No products has been found in rmt. Please run rmt-cli sync before'
+    warn 'importing data from smt.'
+    exit 1
+  end
+end
+
+# Run the script if its called directly from commandline
+# :nocov:
+if $PROGRAM_NAME == __FILE__
```

Comment:
Just move the class into 'lib/' and require it -- and then you don't need those hacks.